### PR TITLE
Added support for extension substatements

### PR DESCRIPTION
**Pull request by apt-itude**
This allows extensions to follow the traditional YIN format in the YINsolidated model by adding the `#yinformat` tag to the extension definition description.

### CODE REVIEW

File: yinsolidated/plugin/plugin.py [hunk 1]
```diff
@@ -288,7 +285,45 @@ def _make_extension_element(statement, parent_elem):
...
+            statement.i_extension
+        )
+        _add_statement_argument(
+            argument_name,
+            statement.arg,
+            namespace,
+            is_arg_yin_element,
+            extension_element,
+        )
+        _append_children(statement, extension_element)
```
Comment (ian-h-chamberlain) [inline yinsolidated/plugin/plugin.py]: Will this recursively add other children (aka can we do deeply nested extensions with this change)? I don't think we have a need at the moment but we may want it in the future.

File: yinsolidated/plugin/plugin.py [hunk 2]
```diff
@@ -288,7 +285,45 @@ def _make_extension_element(statement, parent_elem):
...
+    else:
+        extension_element.text = statement.arg
+
+
+def _is_complex_extension(extension_statement):
+    extension_description = extension_statement.search_one('description')
+    return (
+        extension_description is not None and
+        "#yinformat" in extension_description.arg
```
Comment (ian-h-chamberlain) [inline yinsolidated/plugin/plugin.py]: I'm not super crazy about using the description for this (especially for "public" extensions) but I can't think of anything better at the moment, so... I guess it's fine.

Comment (hadrielk) [reply yinsolidated/plugin/plugin.py]: alternatively we could use the `reference` field, where referencing RFC 6020 or some such would be the trigger.

Comment (ian-h-chamberlain) [reply yinsolidated/plugin/plugin.py]: Ironically, that RFC is _not_ a good reference for the extension substatements. From §7.17.
> The substatements of an extension are defined by the extension, using some mechanism outside the scope of this specification.  Syntactically, the substatements MUST be YANG statements, or also defined using "extension" statements.

File: yinsolidated/plugin/plugin.py (hunk 1 above)
Comment (apt-itude) [reply yinsolidated/plugin/plugin.py]: Yup, this is recursive.

File: yinsolidated/plugin/plugin.py (hunk 2 above)
Comment (apt-itude) [reply yinsolidated/plugin/plugin.py]: Using the `reference` field just doesn't seem intuitive to me. I think it should be obvious when you look at the model that the extension has been tagged with some metadata that's being used programmatically. Just sticking "RFC XXXX" in a reference doesn't make that apparent at all. I also don't think putting the `#yinformat` tag in the reference statement makes much sense. The description is more of a free-form string, so to me, it fits better there. Plus extension descriptions don't end up in being presented on any of our UIs, so it seemed relatively harmless.

**Pull request merged**